lib/mockResponse.js

```diff
@@ -426,15 +432,47 @@ function createResponse(options) {
             }
         }
 
-        if (encoding) {
-            _encoding = encoding;
+        if (args.encoding) {
+            _encoding = args.encoding;
         }
-
+        
         mockResponse.emit('end');
         mockResponse.writableFinished = true; //Reference: https://nodejs.org/docs/latest-v12.x/api/http.html#http_request_writablefinished
         mockResponse.emit('finish');
-
+        
+        if (args.callback) {
+            args.callback();
+        }
     };
+    /**
+     *  Function: end.handleSignatures
+     *
+     *  Utility function that parses and names mockResponse.end various 
+     *  function signatures. Reference:
+     *  https://nodejs.org/api/http.html#http_response_end_data_encoding_callback
+     *  
+     */
+    mockResponse.end.handleSignatures = function (args) {
```

Comment:
What is the benefit of making `handleSignatures()` to be a property of the `mockResponse.end()` instead of making it a simple standalone function like

```
const handleSignatures = function (args) {
 //...
}
```

Reply:
this way the developers know that it belongs to end(), I could declare instead a utility function

`function handleEndSignatures() { [...] }`

which could be used by `request.end()` too, as it has the same signatures as `response.end()`. What do you think? :)

Reply:
@eugef just noticed there is no `end()` method on the mockRequest. Would you prefer I move change `mockResponse.end.handleSignatures` into `handleEndSignatures()` and move it above `mockResponse.end`?

Reply:
yes